satellytes/satellytes.com
# feat: add weather easter egg

### Features
- Added easter egg to aurora hero
- If activated weather data from user location gets fetched
- Weather gets displayed in aurora hero
- If weather is sunny the position of the sun is synced to real life

### Technical information
- Weather data and sunset/sunrise time gets fetched from api.weatherapi.com
- Resturctured aurora inside the hero component:
  1. Aurora types are now in a separate file (aurora-types.ts)
  2. Moved default flares from aurora hero to own component (default-flares.tsx)
  3. Moved AuroraContainer, AuroraBackground and AuroraForeground to aurora-components
- Added weather-easter-egg directory to aurora in hero component with:
  1. weather-api.ts to querry the API
  2. weather-easter-egg.tsx which gets what should be visible at a weather state from flaresByWeather from aurora-components
  3. A comonent for every weather type (cloudy and slightly cloudy are one component)
  4. A custom hook (useWeather) to use the weather API to set a state to the current weather if shortcut is pressed 


### How to verify
1. Open the website
6. Press Shift + Control + Option (Mac) / Shift + Control + Alt (Windows)
7. Current weather should replace the hero


### Todo list
- [ ] add weather api token to 1password (name: WEATHER_API_KEY)

### Screenshots
#### Easter egg disabled:
![image](https://github.com/satellytes/satellytes.com/assets/72336443/009a2f85-ce4f-46db-b6a4-dc0d7bfec90a)

#### Sunny weather:
![image](https://github.com/satellytes/satellytes.com/assets/72336443/c34f866d-40e2-4071-b0cb-3b366fde6bfd)

#### Slightly cloudy weather:
![image](https://github.com/satellytes/satellytes.com/assets/72336443/c26a8972-73a6-4580-9889-f29c54cafcf2)

#### Cloudy weather:
![image](https://github.com/satellytes/satellytes.com/assets/72336443/fd0afa73-991a-40ed-86c7-9744dab953ea)

#### Rainy weather:
![image](https://github.com/satellytes/satellytes.com/assets/72336443/965a89bd-60f6-447e-b0c5-fd0ac09422cf)

#### Snowy weather:
![image](https://github.com/satellytes/satellytes.com/assets/72336443/4f2f32b9-ce34-4422-ba4c-f64d97445275)
